modules/search/class.jetpack-search.php

```diff
@@ -668,7 +668,7 @@ function convert_wp_es_to_es_args( array $args ) {
 			'blog_id'        => get_current_blog_id(),
 
 			'query'          => null,    // Search phrase
-			'query_fields'   => array( 'title', 'content', 'author', 'tag', 'category' ),
+			'query_fields'   => array( 'title', 'content', 'author', 'tag.name', 'category' ),
```

Comment:
Category has the same problem. Should also be `category.name` and be added on the WP.com side.

All taxonomy.*.name fields are the same, but I doubt those will be problems since they require custom coding.